src/Umbraco.Web.UI.Client/src/packages/core/tree/expansion-manager/tree-expansion-manager.ts

```diff
@@ -0,0 +1,71 @@
+import type { UmbTreeExpansionModel } from './types.js';
+import { UmbControllerBase } from '@umbraco-cms/backoffice/class-api';
+import type { UmbEntityModel } from '@umbraco-cms/backoffice/entity';
+import { appendToFrozenArray, UmbObjectState } from '@umbraco-cms/backoffice/observable-api';
+
+/**
+ * Manages the expansion state of a tree
+ * @exports
+ * @class UmbTreeExpansionManager
+ * @augments {UmbControllerBase}
+ */
+export class UmbTreeExpansionManager extends UmbControllerBase {
+	#expansion = new UmbObjectState<UmbTreeExpansionModel | undefined>(undefined);
```

Comment:
The UmbTreeExpansionModel is an Array? so shouldn't this be using the UmbArrayState, cause then line 45 could be using this.#expansion.appendOne() instead, making that code a little simpler.
And the same for 59 which could be this.#expansion.filter(...)

:-)

Reply:
Good catch 👍